sql/parser/builtins.go

```diff
@@ -1161,6 +1186,31 @@ func regexpEvalFlags(pattern, sqlFlags string) (string, error) {
 	return fmt.Sprintf("(?%s:%s)", flagString.String(), pattern), nil
 }
 
+func overlay(s, to string, pos, size int) (Datum, error) {
+	if pos < 1 {
+		return nil, errors.New("negative substring length not allowed")
+	}
+
+	var out bytes.Buffer
+	i := 0
+	for _, c := range s {
```

Comment:
Wouldn't that result in a potentially expensive buffer copy? I think something along these lines was brought up with the implementation of "translate".